staging/src/k8s.io/apiserver/pkg/authentication/cel/compile.go

```diff
@@ -39,6 +40,20 @@ type compiler struct {
 	varEnvs map[string]*environment.EnvSet
 }
 
+var (
+	defaultCompilerOnce sync.Once
+	defaultCompiler     Compiler
+)
+
+// DefaultCompiler returns a shared default Compiler. Do not modify the returned Compiler.
```

Comment:
this must not be called before feature gates and compatibility version are locked, right?

Reply:
right